### PR TITLE
Feature/362 implement new project detail

### PR DESCRIPTION
**Pull request by waltersajtos**
## Description

Implement the modal-design of the project detail page

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Go to the project overview page and click a project
2. A modal should pop up with the project details

## Link to issue
#362 

Closes: #362

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Redesigned project overview page [#365](https://github.com/DigitalExcellence/dex-frontend/issues/365)
+- Redesigned project details apge [#362](https://github.com/DigitalExcellence/dex-frontend/issues/362)
```
Comment (Brend-Smits) [inline CHANGELOG.md:19]:
```suggestion
- Implemented redesigned project details page [#362](https://github.com/DigitalExcellence/dex-frontend/issues/362)
```

File: src/app/modules/project/collaborator/collaborator.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export class CollaboratorComponent implements OnInit {
+
+  @Input() collaborator: Collaborator;
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  public onFollowClick(projectId: number) {
```
Comment (Brend-Smits) [inline src/app/modules/project/collaborator/collaborator.component.ts:19]: Documentation missing

File: src/app/modules/project/collaborator/collaborator.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  }
+
+  public isProduction(): boolean {
```
Comment (Brend-Smits) [inline src/app/modules/project/collaborator/collaborator.component.ts:23]: Documentation missing

File: src/app/modules/project/details/details.component.html [hunk 4]
```diff
@@ -27,51 +27,41 @@
...
     <div class="row">
-      <div class="col-3 project-metadata">
-        <div class="project-icon" style="background-color: unset">
-          <img [src]="getIconUrl()" alt="Code project" loading="lazy" onerror="this.src='/assets/images/placeholder.svg';">
-        </div>
-        <div *ngIf="project.uri" class="project-metadata__item">
-          <h3>External Link</h3>
-
-          <a href="{{ project.uri }}" rel="noreferrer noopener" target="_blank">{{ project.uri }}</a>
+      <!-- <img src="https://www.rover.com/blog/wp-content/uploads/2019/06/bernese-mountain-dog-1177074_1920-960x540.jpg" alt="project-detail-image"> -->
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.html:30]: Commented out code, probably should be removed or explained why it's here

File: src/app/modules/project/details/details.component.scss [hunk 5]
```diff
@@ -18,114 +18,268 @@
...
 }
 
-.project-metadata__item {
-  margin: 30px 0;
-  overflow: hidden;
-  text-overflow: ellipsis;
-  h3,
-  p {
-    margin: 0;
+.modal-dialog{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:26]:
```suggestion
.modal-dialog {
```

File: src/app/modules/project/details/details.component.scss [hunk 6]
```diff
@@ -18,114 +18,268 @@
...
+    @media screen and (max-width: 769px) {
+      grid-template-columns: 1fr 1fr;
+    }
+    .project-metadata__item {
+      overflow: hidden;
+      text-overflow: ellipsis;
+      ul{
+        margin: 0;
+      }
+      .title{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:100]:
```suggestion
      .title {
```

File: src/app/modules/project/details/details.component.scss [hunk 7]
```diff
@@ -18,114 +18,268 @@
...
+        }
+      }
+
+      ul {
+        font-family: "Lato", sans-serif;
+        color: $light-mode-grey-primary;
+        padding: 0;
+      }
+    }
+    .hover{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:131]:
```suggestion
    .hover {
```

File: src/app/modules/project/details/details.component.scss [hunk 8]
```diff
@@ -18,114 +18,268 @@
...
+      &:hover{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:132]:
```suggestion
      &:hover {
```

File: src/app/modules/project/details/details.component.scss [hunk 9]
```diff
@@ -18,114 +18,268 @@
...
+        opacity: 0.75;
+        color: $accent-color-red-primary!important;
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:134]:
```suggestion
        color: $accent-color-red-primary !important;
```

File: src/app/modules/project/details/details.component.scss [hunk 10]
```diff
@@ -18,114 +18,268 @@
...
+        color: $light-mode-black;
+        margin-left: auto;
+        opacity: 0.8;
+        cursor: pointer;
+        font-size: 15px;
+        &:hover{
+          opacity: 1;
+        }
+      }
+      .popover{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:215]:
```suggestion
      .popover {
```

File: src/app/modules/project/details/details.component.scss [hunk 11]
```diff
@@ -18,114 +18,268 @@
...
+        border: none;
+        filter: drop-shadow($box-shadow);
+        .arrow::before{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:218]:
```suggestion
        .arrow::before {
```

File: src/app/modules/project/details/details.component.scss [hunk 12]
```diff
@@ -18,114 +18,268 @@
...
+      }
+    }
+    .tab-content {
+        display: flex;
+        flex-direction: column;
+        gap: 15px;
+      p {
+        margin: 0;
+      }
+      .collaborators{
```
Comment (Brend-Smits) [inline src/app/modules/project/details/details.component.scss:250]:
```suggestion
      .collaborators {
```

**Review by Brend-Smits — CHANGES_REQUESTED** [review]

File: src/app/modules/project/details/details.component.html [hunk 13]
```diff
@@ -27,51 +27,45 @@
...
     <div class="row">
-      <div class="col-3 project-metadata">
-        <div class="project-icon" style="background-color: unset">
-          <img [src]="getIconUrl()" alt="Code project" loading="lazy" onerror="this.src='/assets/images/placeholder.svg';">
-        </div>
-        <div *ngIf="project.uri" class="project-metadata__item">
-          <h3>External Link</h3>
-
-          <a href="{{ project.uri }}" rel="noreferrer noopener" target="_blank">{{ project.uri }}</a>
+      <!-- in this place should be a image or a bootstrap carrousel after the wizzard is finished
```
Comment (juliaslaats) [inline src/app/modules/project/details/details.component.html:30]: English mistakes
"**on this spot** should be **an** image or a bootstrap carrousel after the **wizard** is finished"

**Review by juliaslaats — CHANGES_REQUESTED** [review]
Looks pretty good! I do notice that there's a couple things that are different than on the designs. Not sure if that's on purpose and I missed the memo on it, but maybe you could look at:
- The design says 'contributors' wit a tooltip instead of just 'uploader' on the top info
- The design says 'contributors' instead of 'collaborators' in the bottom menu. 
- The designs have a close button in the right top corner of the screen, that I'm missing here. 
- The designs have an edit-button aligned on the right in the bottom menu, that I'm missing here.

And then there's a couple things I noticed that are not necessarily related to the designs
- The 'uploader' is not included as a collaborator in the menu on the bottom
- The 'give feedback' button jumps when you click on a detail page. 
- The likes are 0 on the overview page, but 120 on the detail page. 

If the differences from the designs were on purpose, or if I had an outdated version of the designs, I apologize. Please let me know 😄

**Comment by liekevdvoort** [comment]
The design says 'contributors' wit a tooltip instead of just 'uploader' on the top info
Yes but i forgot the uploader in the designs and we already have the collaborators in the tab
The design says 'contributors' instead of 'collaborators' in the bottom menu.
Yes but dex already used collaborators so i wanted to keep it consistent
The designs have a close button in the right top corner of the screen, that I'm missing here.
Needs to be added but isn't as easy as thought so i wanted to do this in a new issue
The designs have an edit-button aligned on the right in the bottom menu, that I'm missing here.
you mean the settings this is only showed when it is your project or you are admin

And then there's a couple things I noticed that are not necessarily related to the designs

The 'uploader' is not included as a collaborator in the menu on the bottom
I don't know if they should i think they are not included now
The 'give feedback' button jumps when you click on a detail page.
Can't be prevented because of scroll locking and the scrollbar of the detail modal
The likes are 0 on the overview page, but 120 on the detail page.
The likes are placeholders and won't be shown on production

@juliaslaats

**Comment by waltersajtos** [comment]
Hey, thanks for the pointers. I will answer them inline

> Looks pretty good! I do notice that there's a couple things that are different than on the designs. Not sure if that's on purpose and I missed the memo on it, but maybe you could look at:
> 
> * The design says 'contributors' wit a tooltip instead of just 'uploader' on the top info
- We decided to go with the author there since the contributors are shown in the tab on the bottom.
> * The design says 'contributors' instead of 'collaborators' in the bottom menu.
- We decided to change this because we call it collaborators in the backend as well, if we use different names it becomes confusing.
> * The designs have a close button in the right top corner of the screen, that I'm missing here.
- 
> * The designs have an edit-button aligned on the right in the bottom menu, that I'm missing here.
- I'm not too sure what you mean by the edit-button. If you mean the cog-popOver then you need to make sure that it is actually your project. If you don't have permission to edit it the cog will not be shown.
> 
> And then there's a couple things I noticed that are not necessarily related to the designs
> 
> * The 'uploader' is not included as a collaborator in the menu on the bottom
- If this is an issue it is a backend issue
> * The 'give feedback' button jumps when you click on a detail page.
- This happens because the scroll bar moves up to the modal, if this is a big problem we need to make a hacky solution which I don't prefer.
> * The likes are 0 on the overview page, but 120 on the detail page.
- This made the styling easier, the likes are hidden in production until we complete the implementation of them so this shouldn't be an issue
> 
> If the differences from the designs were on purpose, or if I had an outdated version of the designs, I apologize. Please let me know 😄

**Review by juliaslaats — APPROVED** [review]
Thanks for addressing my comments! It looks good to me 👍

File: src/app/modules/project/overview/overview.component.ts [hunk 14]
```diff
@@ -358,4 +386,27 @@ export class OverviewComponent implements OnInit {
...
+            this.updateSEOTags();
+          }));
+    }
+  }
+
+  private updateSEOTags() {
+    // Updates meta and title tags
+    this.seoService.updateTitle('Project overview');
+    this.seoService.updateDescription('Browse or search for specific projects or ideas within DeX');
+  }
```
Comment (Ruby77) [inline src/app/modules/project/overview/overview.component.ts:411]: I would suggest adding documentation to these methods as well.

**Review by Ruby77 — CHANGES_REQUESTED** [review]
Looks really nice again 😄.

One small question:
image
When I tried to close the settings, I clicked somewhere on the screen expecting the popup would hide (just like for the sign out menu). I noticed it only hides whenever I clicked on the settings button. Is this a specific choice, why it won't hide whenever I click somewhere else?

**Comment by waltersajtos** [comment]
> Looks really nice again 😄.
> 
> One small question:
> image
> When I tried to close the settings, I clicked somewhere on the screen expecting the popup would hide (just like for the sign out menu). I noticed it only hides whenever I clicked on the settings button. Is this a specific choice, why it won't hide whenever I click somewhere else?

I switched from a tooltip to a popover because tooltips do not allow the anchor tags. I forgot to check the behavior, nice catch!

**Review by Ruby77 — APPROVED** [review]
Nice 🎉

**Review by Brend-Smits — APPROVED** [review]
Nice work everyone!

**Pull request merged**